### PR TITLE
[TextLogging] Allow custom logger name

### PR DESCRIPTION
**Pull request by RiccardoGrieco**
When using the implemented Logger Factories, the logger's name (currently `blf`) is displayed in the formatted message -  see [here](https://github.com/ami-iit/bipedal-locomotion-framework/blob/6747c2dfb0f3fc3f1feeb94ad911095ccb356780/src/TextLogging/YarpImplementation/src/YarpLogger.cpp#L42) for a reference (where `%n` is the name of the logger).

Users may not want to use `blf`, so this PR to allow the usage of a custom name for the implemented loggers of the TextLogging component.

The changes do not break the current API since the default name `blf` is provided as default value.

### CODE REVIEW

File: src/TextLogging/YarpImplementation/include/BipedalLocomotion/TextLogging/YarpLogger.h [hunk 1]
```diff
@@ -88,11 +88,20 @@ using YarpSink_mt = YarpSink<std::mutex>;
 class YarpLoggerFactory final : public LoggerFactory
 {
 public:
+    /**
+     * Construct a new YarpLoggerFactory object
+     * @param name the name of the logger which will be used inside the formatted messages
+     */
+    YarpLoggerFactory(const std::string_view& name = "blf") : m_name{name} {}
```
Comment (S-Dafarra) [inline src/TextLogging/YarpImplementation/include/BipedalLocomotion/TextLogging/YarpLogger.h:95]: Could you move the implementation in the cpp file?

File: src/TextLogging/include/BipedalLocomotion/TextLogging/DefaultLogger.h [hunk 2]
```diff
@@ -18,11 +19,20 @@ namespace TextLogging
 class DefaultLoggerFactory final : public LoggerFactory
 {
 public:
+    /**
+     * Construct a new DefaultLoggerFactory object
+     * @param name the name of the logger which will be used inside the formatted messages
+     */
+    DefaultLoggerFactory(const std::string_view& name = "blf") : m_name{name}{}
```
Comment (S-Dafarra) [inline src/TextLogging/include/BipedalLocomotion/TextLogging/DefaultLogger.h:26]: Same here

File: src/TextLogging/YarpImplementation/src/YarpLogger.cpp [hunk 3]
```diff
@@ -44,6 +44,8 @@ std::shared_ptr<TextLogging::Logger> _createLogger(const std::string& name)
     return logger;
 }
 
+TextLogging::YarpLoggerFactory::YarpLoggerFactory(const std::string_view& name) : m_name{name} {}
```
Comment (GiulioRomualdi) [inline src/TextLogging/YarpImplementation/src/YarpLogger.cpp:47]: could you please apply clang-format? Thanks :)

**Review by GiulioRomualdi — COMMENTED** [review]
minor comment

**Comment by GiulioRomualdi** [comment]
Thank you @RiccardoGrieco

**Pull request merged**